arch/arm/boot/dts/imx6ul-ts7100.dtsi

```diff
@@ -0,0 +1,530 @@
+	bus-width = <4>;
+	status = "okay";
+};
+
+&wdog1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_wdog>;
+	status = "okay";
+	fsl,ext-reset-output;
+};
+
+&weim {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_weim_fpga>;
+	clocks = <&clks IMX6UL_CLK_EIM>, <&clks IMX6UL_CLK_EIM_SLOW_SEL>;
+	ranges = <0 0 0x50000000 0x08000000>;
+	status = "okay";
+
+	#address-cells = <2>;
+	#size-cells = <1>;
+
+	fpga: fpgabus@50000000 {
+		compatible = "simple-bus";
+		reg = <0 0x50000000 0x00010000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges = <0 0 0 0x10000>;
+
+		fsl,weim-cs-timing = <
+			0x0161030F	// EIM_CSnGCR1 @ 0x021b8000
+			0x00000000	// EIM_CSnGCR2 @ 0x021b8004
+			0x03000000	// EIM_CSnRCR1 @ 0x021b8008
+			0x00000000	// EIM_CSnRCR2 @ 0x021b800c
+			0x01000000	// EIM_CSnWCR1 @ 0x021b8010
+			0		// EIM_CSnWCR2 @ 0x021b8014
+		>;
+
+		fpga_clk_weim_bclk: fpga_clk_weim_bclk {
+			#clock-cells = <0>;
+			compatible = "fixed-clock";
+			clock-frequency = <49500000>;
+		};
+
+		syscon: syscon@50004000 {
+			compatible = "simple-bus";
+			#address-cells = <0x2>;
+			#size-cells = <0x1>;
+			reg = <0x4000 0x58>;
+			ranges = <0 0 0x4000 0x58>;
+
+			gpio6: gpio@50004010 {
+				compatible = "technologic,ts71xxweim-gpio";
+				reg = <0 0x10 0x08>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				#interrupt-cells = <1>;
+			};
+
+			gpio7: gpio@50004040 {
+				compatible = "technologic,ts71xxweim-gpio";
+				reg = <0 0x40 0x08>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				#interrupt-cells = <1>;
+			};
+
+			gpio8: gpio@50004050 {
+				compatible = "technologic,ts71xxweim-gpio";
+				reg = <0 0x50 0x08>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				#interrupt-cells = <1>;
+			};
+
+			fpga_intc: interrupt-controller@50004024 {
+				compatible = "technologic,ts71xxweim-intc";
+
+				interrupt-controller;
+				#interrupt-cells = <1>;
+				reg = <0 0x00 0x50>;
+
+				interrupt-parent = <&gpio5>;
+				interrupts = <1 IRQ_TYPE_LEVEL_HIGH>;
+			};
+		};
+
+		fpga_uart0: serial@50000000 {
+			compatible = "ns16550a";
+			device_type = "serial";
+			current-speed = <115200>;
+			reg-shift = <1>;
+			reg-io-width = <1>;
+			reg = <0 16>;
+			clock-frequency = <1843200>;
+			interrupt-parent = <&fpga_intc>;
+			interrupts = <0>;
+		};
+
+		opencores_spi0: spi@50000100 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			compatible = "opencores,spi-oc";
+			reg = <0x100 32>;
+			interrupt-parent = <&fpga_intc>;
+			interrupts = <9>;
+			clocks = <&fpga_clk_weim_bclk>;
+			clock-names = "spi-oc-clk";
+			opencores-spi,idx = <0>;
+			opencores-spi,num-chipselects = <2>;
+
+			spifram: eeprom@0 {
+				compatible = "atmel,at25";
+				reg = <0>;
+				spi-max-frequency = <20000000>;
+				size = <0x800>;
+				address-width = <16>;
+				pagesize = <64>;
+			};
+
+			spisplash: spi@1 {
+				compatible = "sst,is25lq016b", "jedec,spi-nor";
```

Comment:
From `Documentation/devicetree/bindings/mtd/jedec,spi-nor.txt`

- compatible : **May** include a device-specific string consisting of the
               manufacturer and name of the chip. **A list of supported chip**
               **names follows.**
               Must also include "jedec,spi-nor" for any SPI NOR flash that can
               be identified by the JEDEC READ ID opcode (0x9F).


The is25lq016b isn't listed as one of the compatible names. I _think_ it should be removed in favor of just `jedec,spi-nor` but I'm not 100% on that.